open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+                if (!(Boolean.parseBoolean(fromDateTimeValidation.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                    return fromDateTimeValidation;
+                }
+
+                String validFrom = controlParameters.getAsString(ConsentExtensionConstants.VALID_FROM_DATE_TIME);
+                OffsetDateTime validFromDateTime = OffsetDateTime.parse(validFrom);
+                OffsetDateTime currentDateTime = OffsetDateTime.now(validToDateTime.getOffset());
+
+                // If ValidToDateTime is older than current date OR ValidToDateTime is older than ValidFromDateTime,
+                // return error
+                if (!validFromDateTime.isBefore(currentDateTime) || !currentDateTime.isBefore(validToDateTime)) {
+                    log.error(String.format("Invalid date-time range, " +
+                                    "validToDateTime: %s, validFromDateTime: %s, currentDateTime: %s",
+                            validToDateTime, validFromDateTime, currentDateTime));
+
+                    String errorMessage = String.format(ErrorConstants.DATE_INVALID_PARAMETER_MESSAGE,
+                            validateToDateTime, validateFromDateTime, currentDateTime);
+
+                    return ConsentManageUtil.getValidationResponse(errorMessage);
+                }
+            } else {
+                // If ValidFromDateTime key is not present, return error
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.MISSING_VALID_FROM_DATE_TIME);
+            }
+        } else {
+            // If ValidToDateTime key is not present, return error
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.MISSING_VALID_TO_DATE_TIME);
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Validator class to validate the payload of a variable recurring payment initiation.
+     * This method performs validation on the initiation payload for a variable recurring payment.
+     * It checks and validates the debtor account and creditor account information if present in the payload.
+     * If any validation fails, it returns a JSON object with details about the validation error.
+     * If the initiation payload passes all validations, the returned JSON object indicates a valid initiation.
+     *
+     * @param initiation The JSON object representing the variable recurring payment initiation payload.
+     * @return validationResponse
+     */
+    public static JSONObject validateVRPInitiationPayload(JSONObject initiation) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        //Validate DebtorAccount
+        if (initiation.containsKey(ConsentExtensionConstants.DEBTOR_ACC)) {
+
+            Object debtorAccount = initiation.get(ConsentExtensionConstants.DEBTOR_ACC);
+
+            String errorMessage = String.format(ErrorConstants.INVALID_PARAMETER_MESSAGE,
```

Comment:
We dont need to create errorMessage variable here right? this is only needed if the validation is false, Therefor shall we move this inside to check block ? (L403)

Reply:
Shall we check and update this in other places as well?

Reply:
Fixed in [e8d8048]